### ISSUE TITLE
Enhancement: setState method functioning

### ISSUE THREAD

**Issue by EliasDalvite** [Pyozer/introduction_screen#183]
**The idea:**
The initial idea was to use the dependency as a configuration step, where I would use Shared Preferences or Hive to store the user's date.

**Problem:**
However, when using a Radio Button, it needs the functionality of setState, and I saw that it is not working when it comes to rendering the widget again whenever setState is run.

**Describe the solution you'd like**
Of course, the solution would be to the setState to start to work... I don't know if it **Isn't** intended to work with the setState method.

**Additional context**
Unfortunately I'm not that experienced with Flutter yet so I can't think about a way to fix that, if it's the case.
From my perspective, enabling this feature to the dependency would improve a lot in how many ways you can use it. But again, I didn't find any information about it being, in fact, not intended to work, to help in performance or something like that.

**Package version or git hash**
- Version of IntroductionScreen [3.1.6]

**Comment by Macacoazul01** [comment]
Can you send a sample code?

**Comment by EliasDalvite** [comment]
So, yesterday I tested the same Widget in a new project and it didn't work.
Now that i saw your answer I started reducing the code to send only the problem over here, and guess what, it worked!.
After some time I noticed the error.
The problem had to do in the way I declared the pages list. 
Declared it as a later list -> after it I assigned to it value from a function (list of pages) in the initiState() -> then I gave its value to the IntroductionScreen pages list.
Removing the variable and assigning the function directly to the pages value worked well, so the problem might be related to the late modifier on the variable as I started it. 

Anyway, thanks for answering my question! E bom final de semana ahahahaha!
